exporter/observiqexporter/converter_test.go

```diff
@@ -15,14 +15,229 @@
+			func() pdata.LogRecord {
+				logRecord := pdata.NewLogRecord()
+				logRecord.Body().SetStringVal("Message")
+				logRecord.Attributes().InsertString(conventions.AttributeServiceName, "myapp")
+				logRecord.Attributes().InsertString(conventions.AttributeHostName, "myhost")
+				logRecord.Attributes().InsertString("custom", "custom")
+				logRecord.SetTimestamp(nanoTs)
+				return logRecord
+			},
+			pdata.NewResource,
+			"agent",
+			"agentID",
+			observIQLogEntry{
+				Timestamp: stringTs,
+				Message:   "Message",
+				Severity:  "default",
+				Resource:  map[string]interface{}{},
+				Data: map[string]interface{}{
+					strings.ReplaceAll(conventions.AttributeServiceName, ".", "_"): "myapp",
+					strings.ReplaceAll(conventions.AttributeHostName, ".", "_"):    "myhost",
+					"custom": "custom",
+				},
+				Agent: &observIQAgentInfo{Name: "agent", ID: "agentID", Version: "latest"},
+			},
+			false,
+		},
+		{
+			"works with attributes of all types",
+			func() pdata.LogRecord {
+				logRecord := pdata.NewLogRecord()
+				logRecord.Body().SetStringVal("Message")
+				logRecord.Attributes().InsertString(conventions.AttributeServiceName, "myapp")
+				logRecord.Attributes().InsertBool("bool", true)
+				logRecord.Attributes().InsertDouble("double", 1.0)
+				logRecord.Attributes().InsertInt("int", 3)
+				logRecord.Attributes().InsertNull("null")
+
+				mapVal := pdata.NewAttributeValueMap()
+				mapVal.MapVal().Insert("mapKey", pdata.NewAttributeValueString("value"))
+				logRecord.Attributes().Insert("map", mapVal)
+
+				arrVal := pdata.NewAttributeValueArray()
+				arrVal.ArrayVal().Resize(2)
+				arrVal.ArrayVal().At(0).SetIntVal(1)
+				arrVal.ArrayVal().At(1).SetIntVal(2)
+				logRecord.Attributes().Insert("array", arrVal)
+
+				logRecord.SetTimestamp(nanoTs)
+				return logRecord
+			},
+			pdata.NewResource,
+			"agent",
+			"agentID",
+			observIQLogEntry{
+				Timestamp: stringTs,
+				Message:   "Message",
+				Severity:  "default",
+				Resource:  map[string]interface{}{},
+				Data: map[string]interface{}{
+					strings.ReplaceAll(conventions.AttributeServiceName, ".", "_"): "myapp",
+					"bool":   true,
+					"double": float64(1.0),
+					"int":    float64(3),
+					"map": map[string]interface{}{
+						"mapKey": "value",
+					},
+					"array": []interface{}{float64(1), float64(2)},
+				},
+				Agent: &observIQAgentInfo{Name: "agent", ID: "agentID", Version: "latest"},
+			},
+			false,
+		},
+		{
+			"Body is nil",
+			func() pdata.LogRecord {
+				logRecord := pdata.NewLogRecord()
+				logRecord.SetTimestamp(nanoTs)
+				return logRecord
+			},
+			pdata.NewResource,
+			"agent",
+			"agentID",
+			observIQLogEntry{
+				Timestamp: stringTs,
+				Message:   "",
+				Severity:  "default",
+				Resource:  map[string]interface{}{},
+				Data:      nil,
+				Agent:     &observIQAgentInfo{Name: "agent", ID: "agentID", Version: "latest"},
+			},
+			false,
+		},
+		{
+			"Body is map",
```

Comment:
The reason why I don't `[]byte` is because as far as I know, there's no way for that type to be represented on pdata.LogRecord -- the AttributeValue struct seems to only support null, string, int, double, bool, map, and array. Maybe this is more of a concern earlier in the pipeline? Otherwise, I'd need help figuring out how to represent this case.

Reply:
I see that it was added to the [proto](https://github.com/open-telemetry/opentelemetry-proto/pull/297/files), but hasn't yet been incorporated into the collector. We can worry about this later.